src/Core/Form/ConfigurableFormChoiceProviderInterface.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Form;
+
+/**
+ * Interface for services that provide configurable form choices (e.g. States choices depending on country).
+ */
+interface ConfigurableFormChoiceProviderInterface
+{
+    /**
+     * @param array $options
+     *
+     * @return array
+     */
+    public function getChoices(array $options);
```

Comment:
Mmmm 🤔 I see why you introduced this. So this is a opponent of `FormChoiceProviderInterface`.
Now that I see it I wonder why we did not introduce it sooner.

@mickaelandrieu @sarjon @tomas862 wdyt ?

Reply:
I'm worried there is somewhere some generic code, like a Factory, expecting a `FormChoiceProviderInterface`.

Reply:
What do you mean @matks?

For choices that don't need any options, we keep using `FormChoiceProviderInterface`, however for choices that can be configured, we can use `ConfigurableFormChoiceProviderInterface` instead.